src/selector.rs

```diff
@@ -13,35 +17,41 @@ use ElementRef;
 #[derive(Debug, Clone, PartialEq)]
 pub struct Selector {
     /// The CSS selectors.
-    pub selectors: Vec<parser::Selector<Simple>>,
+    pub selectors: SmallVec<[parser::Selector<Simple>; 1]>,
 }
 
 impl Selector {
     /// Parses a CSS selector group.
-    ///
-    /// No meaningful error can be returned here, due to a limitation of the `selectors` and
-    /// `cssparser` crates.
-    pub fn parse(selectors: &str) -> Result<Self, ()> {
-        let mut parser = cssparser::Parser::new(selectors);
+
+    pub fn parse<'t, 'i>(
+        selectors: &'i str,
+    ) -> Result<Self, cssparser::ParseError<'t, SelectorParseErrorKind<'i>>> {
+        let mut _selectors = cssparser::ParserInput::new(selectors);
```

Comment:
This probably shouldn't be named with a leading `_` since that indicated an unused variable. It could actually just be named `selectors` and shadow the argument name.